### PR TITLE
HIP-584: precompile pricing logic

### LINKED ISSUE #5666: HIP-584: Copy precompile pricing logic without `gasFeeInTinybars` into hedera-mirror-node
### Problem

We need to extract the needed logic for calculating the gas cost of each precompile.

### Solution

Copy `PrecompilePricingUtils` and its associated logic without `gasFeeInTinybars` from `hedera-services` to `hedera-mirror-node`. `gasFeeInTinybars` is needed only for `TokenCreatePrecompile` to calculate the hbars we charge in addition to the gas fee. This issue aims to cover the main precompile pricing logic that is common for most precompiles. 

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by 0xivanov**
**Description**:

**Related issue(s)**:

Fixes #5666

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/calculation/usage/consensus/ConsensusOpsUsage.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import static com.hedera.services.hapi.utils.fees.FeeBuilder.TX_HASH_SIZE;
+
+import com.hedera.services.fees.usage.state.UsageAccumulator;
+import com.hedera.services.hapi.fees.usage.BaseTransactionMeta;
+import com.hedera.services.hapi.fees.usage.SigUsage;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public final class ConsensusOpsUsage {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/fees/calculation/usage/consensus/ConsensusOpsUsage.java:31]: We won't need this class.

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/calculation/usage/consensus/SubmitMessageMeta.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.fees.calculation.usage.consensus;
+
+public record SubmitMessageMeta(int numMsgBytes) {}
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/fees/calculation/usage/consensus/SubmitMessageMeta.java:19]: We won't need this record.

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/pricing/AssetsLoader.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    /**
+     * Loads a map that, for each resource type available in the network, gives the "capacity" of
+     * that resource. These capacities do not have an absolute meaning, and are just compared to
+     * infer the relative scarcity of each resource.
+     *
+     * @return the network "capacity" of each resource type
+     * @throws IOException if the backing JSON resource cannot be loaded
+     */
+    Map<UsableResource, BigDecimal> loadCapacities() throws IOException {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/fees/pricing/AssetsLoader.java:80]: We won't need this method.

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/pricing/AssetsLoader.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    /**
+     * Loads a map that, for each supported operation, gives the desired price in USD for the "base
+     * configuration" of each type of that operation. (Types are given by the values of the {@link
+     * SubType} enum; that is, DEFAULT, TOKEN_NON_FUNGIBLE_UNIQUE, and TOKEN_FUNGIBLE_COMMON.)
+     *
+     * @return the desired per-type prices, in USD
+     * @throws IOException if the backing JSON resource cannot be loaded
+     */
+    public Map<HederaFunctionality, Map<SubType, BigDecimal>> loadCanonicalPrices() throws IOException {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/fees/pricing/AssetsLoader.java:109]: We should use a similar approach to BasicFcfsUsagePrices.defaultPricesGiven and reuse the way we fetch FeeSchedules from DB and then reconstruct the fetched data in this type -> `Map<HederaFunctionality, Map<SubType, BigDecimal>` as it's done in this method.

In other words, we can't rely on a json file under `CANONICAL_PRICES_RESOURCE` but we should use the DB.

File: hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+        totalXfers += xferMeta.getCustomFeeHbarTransfers();
+        weightedTokenXfers += tokenMultiplier * xferMeta.getCustomFeeTokenTransfers();
+        weightedTokensInvolved += tokenMultiplier * xferMeta.getCustomFeeTokensInvolved();
+        long incRb = totalXfers * LONG_ACCOUNT_AMOUNT_BYTES;
+        incRb += TOKEN_ENTITY_SIZES.bytesUsedToRecordTokenTransfers(
+                weightedTokensInvolved, weightedTokenXfers, xferMeta.getNumNftOwnershipChanges());
+        accumulator.addRbs(incRb * USAGE_PROPERTIES.legacyReceiptStorageSecs());
+    }
+
+    public FeeData cryptoInfoUsage(final Query cryptoInfoReq, final ExtantCryptoContext ctx) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java:92]: We don't need this method.

File: hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+        final var op = cryptoInfoReq.getCryptoGetInfo();
+
+        final var estimate = queryEstimateFactory.apply(op.getHeader().getResponseType());
+        return getUsage(estimate, ctx);
+    }
+
+    public FeeData accountDetailsUsage(final Query accountDetailsReq, final ExtantCryptoContext ctx) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java:99]: We don't need this method.

File: hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+        final var op = accountDetailsReq.getAccountDetails();
+
+        final var estimate = queryEstimateFactory.apply(op.getHeader().getResponseType());
+        return getUsage(estimate, ctx);
+    }
+
+    private FeeData getUsage(final QueryUsage estimate, final ExtantCryptoContext ctx) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java:106]: We don't need this method.

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/calculation/usage/consensus/ConsensusOpsUsage.java (hunk 1 above)
Comment (ivan-02) [reply hedera-mirror-web3/src/main/java/com/hedera/services/fees/calculation/usage/consensus/ConsensusOpsUsage.java:28]: Done

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/calculation/usage/consensus/SubmitMessageMeta.java (hunk 2 above)
Comment (ivan-02) [reply hedera-mirror-web3/src/main/java/com/hedera/services/fees/calculation/usage/consensus/SubmitMessageMeta.java]: Done

File: hedera-mirror-web3/src/main/java/com/hedera/services/fees/pricing/AssetsLoader.java (hunk 3 above)
Comment (ivan-02) [reply hedera-mirror-web3/src/main/java/com/hedera/services/fees/pricing/AssetsLoader.java]: Done

File: hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java (hunk 5 above)
Comment (ivan-02) [reply hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java]: Done

File: hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java (hunk 7 above)
Comment (ivan-02) [reply hedera-mirror-web3/src/main/java/com/hedera/services/hapi/fees/usage/crypto/CryptoOpsUsage.java]: Done

**Pull request closed without merging**